packages/shortest/src/cache/cache.ts

```diff
@@ -0,0 +1,219 @@
+import * as fs from "fs";
+import path from "path";
+import { CacheEntry, CacheStore } from "../types/cache";
+import { hashData } from "../utils/crypto";
+import { Logger } from "../utils/logger";
+import * as objects from "../utils/objects";
+
+export class BaseCache<T extends CacheEntry> {
+  private readonly CACHE_TTL = 7 * 24 * 60 * 60 * 1000; // 1 week
+  private readonly CLEANUP_PROBABILITY = 0.03; // 3% chance
+
+  private logger: Logger;
+
+  private cacheFile: string;
+
+  // Locking (ensures that only one process or thread can access or modify cache)
+  private lockFile: string;
+  private readonly LOCK_TIMEOUT_MS = 1_000;
+  protected lockAcquired = false;
+  protected lockAcquireFailures = 0;
+
+  constructor() {
+    this.logger = new Logger();
+    this.cacheFile = path.join(process.cwd(), ".cache", "cache.json");
+    this.lockFile = path.join(process.cwd(), ".cache", "cache.lock");
```

Comment:
I think it would be better to put this inside `.shortest` folder. Also does this only cache passing test cases?

Reply:
> I think it would be better to put this inside .shortest folder.

Good point. I will rename it

> Also does this only cache passing test cases?

Not entirely sure what you mean. If you are asking whether test file only contains tests that succeeded, then yes